### ISSUE TITLE
Feature: Parameters: add support for manual value refresh (WIP)

### ISSUE THREAD

**Issue by nmoutschen** [awslabs/aws-lambda-powertools-python#131]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

This would add a method to the parameter provider to force-refresh a parameter value. This could be done through a flag.

Use cases: username/password that have expired and need to be refreshed.

**Describe alternatives you've considered**

**Additional context**

**Comment by michaelbrewer** [comment]
@nmoutschen @heitorlessa would this be via some kind of exception handler that picks up when the credentials failed?

**Comment by michaelbrewer** [comment]
@nmoutschen  a version of this is in this PR : https://github.com/awslabs/aws-lambda-powertools-python/pull/341

UX might be better for a decorator.

**Comment by heitorlessa** [comment]
@nmoutschen could you expand on what you meant with this feature request so we can get it added in Wednesday's release?

Do you mean a parameter like what @michaelbrewer implemented on a per `get_parameter` call, or an explicit method to refresh what's already in the cache e.g. `.refresh_parameter(name='/dev/service/parameter')`?

If it's the former, @michaelbrewer I'd suggest a more explicit name like `ignore_cache` as that's what it seems to be happening in the implementation.

**Comment by nmoutschen** [comment]
Hey @michaelbrewer & @heitorlessa!

I didn't have a specific implementation in mind and thought of 2 possibilities: pass a flag to the `get_parameter` call, or manually flush a specific cache value. Overall, I think @michaelbrewer's implementation is simpler (same function, with just an additional parameter).

From a name point of view, `ignore_cache` would imply that the parameter wouldn't be stored in local cache, so I personally prefer `force_update`. `force_fetch` is good too.

**Comment by heitorlessa** [comment]
Thanks @nmoutschen that makes sense. `force_fetch` in this case makes more sense -- @michaelbrewer could you do a quick refactor on that PR with that and we can merge ahead of Wed's release?

**Comment by michaelbrewer** [comment]
@heitorlessa  i have made the changes just thinking of a good example for the docs, here it was i have so far:

```python3
import os
from aws_lambda_powertools.utilities import parameters


class InvalidCredentials(Exception):
    """Raised when client get an invalid credentials error"""


class Client:
    password = None

    def configure(self, password):
        self.password = password

    def read_record(self):
        """Might raise an InvalidCredentials exception """


DB_PASS_PARAM = os.environ["DB_PASS_PARAM"]
db_client = Client()
db_client.configure(parameters.get_parameter(DB_PASS_PARAM))


def refresh_db_password():
    db_client.configure(parameters.get_parameter(DB_PASS_PARAM, force_fetch=True))


def read_record(is_retry=False):
    try:
        return db_client.read_record()
    except InvalidCredentials:
        if not is_retry:  # avoid infinite recursion
            refresh_db_password()  # force parameter refresh
            return read_record(is_retry=True)


def handler(event, context):
    return {"record": read_record()}

```

**Comment by heitorlessa** [comment]
This is now available in 1.12.0 :) Thanks @michaelbrewer ! 

https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.12.0
